filesystem/dir.c

```diff
@@ -57,21 +57,25 @@ void FS_Search_DIR( searchpath_t *search, stringlist_t *list, const char *patter
 	slash = Q_strrchr( pattern, '/' );
 	backslash = Q_strrchr( pattern, '\\' );
 	colon = Q_strrchr( pattern, ':' );
+
 	separator = Q_max( slash, backslash );
 	separator = Q_max( separator, colon );
+	
 	basepathlength = separator ? (separator + 1 - pattern) : 0;
 	basepath = Mem_Calloc( fs_mempool, basepathlength + 1 );
 	if( basepathlength ) memcpy( basepath, pattern, basepathlength );
-	basepath[basepathlength] = 0;
+	basepath[basepathlength] = '\0';
 
-	Q_sprintf( netpath, "%s%s", search->filename, basepath );
+	Q_strcpy( temp,  basepath );
```

Comment:
What are you doing?

Reply:
meth

Reply:
Actually I can't find this code, so not sure where did you get that?